### ISSUE TITLE
Tag versions and hugo modules

### ISSUE THREAD

**Issue by aljaznuncic** [mfg92/hugo-shortcode-gallery#64]
It will be very good, if we start versioning code with tags, so we can use this shortcode with hugo modules: https://gohugo.io/hugo-modules/use-modules/#get-a-specific-version

**Comment by mfg92** [comment]
Good point @aljaznuncic. I have now finally released version [1.0.0](https://github.com/mfg92/hugo-shortcode-gallery/releases/tag/1.0.0) of hugo-shortcode-gallery.
Can hugo-shortcode-gallery actually be used as a hugo module now? I have not used any hugo modules for my own website, so I am a newbie in this area. I did a quick search and it seems that at least a _hugo.yaml_ file is needed in hugo-shortcode-gallery to make it work as a hugo module.
Can you give me any feedback on this? A PR is also very welcome.

**Comment by sheldonhull** [comment]
Go based tagging scheme requires `v` prefix in the tag name and since adding a new module is using hugos version of go mod I believe, not sure it will work without that prefix. Will give it a shot.

**Comment by mfg92** [comment]
I have added the "v", can you try again?

**Comment by josephbadow** [comment]
Let's warm this up, I'm interested as well. I just tested it on my machine, but I have no experience with hugo modules so far. Here is what I did:

Remove theme reference from my config. In my case that means removing the whole line
```
theme = ["hugo-shortcode-gallery"]
```
I also renamed the folder `themes/hugo-shortcode-gallery` for good measure.

Adding hugo-shortcode-gallery as new item in [module.imports](https://gohugo.io/hugo-modules/configuration/#module-configuration-imports)
```
[[imports]]
path = "github.com/mfg92/hugo-shortcode-gallery"
```
"Downloading" the module
```
hugo mod get github.com/mfg92/hugo-shortcode-gallery@latest
```

Now, when I run `hugo mod graph` I see both modules:
```
hugo mod graph 
github.com/jpanther/congo github.com/jpanther/congo/v2@v2.8.0
github.com/jpanther/congo github.com/mfg92/hugo-shortcode-gallery@v1.0.0
```
Running `hugo` simply renders everything without errors. I only have one post with the gallery so far, so there is not much that I can test at the moment. (I'm using the [congo](https://github.com/jpanther/congo/) theme, as you can see.)

**Comment by mfg92** [comment]
@josephbadow I have finally set up hugo-shortcode-gallery as a Hugo module. Could you please test it to ensure it works as expected? Although it worked on my end, it would be great to have another perspective :)

I have created a release wit this change: [v1.1.0](https://github.com/mfg92/hugo-shortcode-gallery/releases/tag/v1.1.0)

Feel free to share your thoughts on the updated [README.md](https://github.com/mfg92/hugo-shortcode-gallery?tab=readme-ov-file#method-a---install-as-hugo-module) too. Thank you!

**Comment by josephbadow** [comment]
I started by updating my modules (I think that command is currently missing in your readme) and that worked flawlessly.

```bash
# Displaying currently installed modules
$ hugo mod graph
github.com/jpanther/congo github.com/jpanther/congo/v2@v2.8.0
github.com/jpanther/congo github.com/mfg92/hugo-shortcode-gallery@v1.0.0

# Updating modules
$ hugo mod get -u
go: downloading github.com/mfg92/hugo-shortcode-gallery v1.1.0
go: upgraded github.com/mfg92/hugo-shortcode-gallery v1.0.0 => v1.1.0

# Building pages
$ hugo
WARN  Failed to read module config for "github.com/mfg92/hugo-shortcode-gallery" in "/home/joseph/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/mfg92/hugo-shortcode-gallery@v1.1.0/theme.toml": "_stream.toml:8:20": unmarshal failed: toml: float can have at most one decimal point
Start building sites … 
hugo v0.122.0-b9a03bd59d5f71a529acb3e33f995e0ef332b3aa+extended linux/amd64 BuildDate=2024-01-26T15:54:24Z VendorInfo=snap:0.122.0


                   | EN   
-------------------+------
  Pages            | 100  
  Paginator pages  |   5  
  Non-page files   |  31  
  Static files     |  24  
  Processed images | 214  
  Aliases          |  35  
  Sitemaps         |   1  
  Cleaned          |   0  

Total in 449 ms
```

I received a warning during build, which I saw at other points as well (https://github.com/gohugoio/hugo/issues/11558, https://github.com/golang/go/issues/61888). It doesn't affect my page and the one gallery I have is still behaving as expected.

**Comment by mfg92** [comment]
That warning is now fixed thanks to #69.

**Comment by mfg92** [comment]
I have incorporated your feedback about the readme in the [latest commit](https://github.com/mfg92/hugo-shortcode-gallery/commit/7d1b5d03eddf7d5d576165b6d235b348977ee33a).
I will close this issue now. If you have any problems, feel free to post here or open a new issue.
